src/main/java/org/spongepowered/api/data/key/Key.java

```diff
@@ -66,4 +74,23 @@
      */
     DataQuery getQuery();
 
+    interface Builder<E, V extends BaseValue<E>> extends ResettableBuilder<Key<V>, Builder<E, V>> {
+
+        <T, B extends BaseValue<T>> Builder<T, B> valueElement(TypeToken<T> element, TypeToken<B> token);
+
+        Builder<E, V> keyId(String id);
```

Comment:
Just `id`?

Reply:
We should be more consistent with builder methods for catalog types:
https://github.com/SpongePowered/SpongeAPI/issues/1721